ocaml/testsuite/tests/comprehensions/syntax.ml

```diff
@@ -44,18 +38,28 @@ let test_printing () =
       parsed printed
 ;;
 
-let () = test_printing ();;
+let () =
+  (* The wonky formatting here is the best way to keep the line
+     break visible in the string and also keep things within 80
+     characters here *)
+ test_printing
+    "[(i, j)\n\
+    \  for i = 0 to 9 when (i mod 2) = 0 for j = 0 to i when (i > 4) && (j > 4)]";
+ (* Check that only user attributes survive pprintast *)
+ test_printing
+   "(([((x)[@attr1 ]) for ((x)[@attr2 ]) in (([])[@attr3 ])])[@attr4 ])";
+;;
 
 (******************************************************************************)
 
 this_test_tests_that
-  "compiler-generated extension nodes start with \"extension.\""
+  "compiler-generated attributes start with \"jane.\""
 ;;
 
 (* In particular: As long as we're in the Jane Street compiler, the compiler
-   translates list comprehensions to an AST in terms of %-annotated extension
-   nodes.  We want to confirm that we're only touching ones we intend to use,
-   which we mark by beginning them with the string "extension.".  We print out
+   translates list comprehensions to an AST in terms of attributes.  We want to
+   confirm that we're only touching ones we intend to use,
+   which we mark by beginning them with the string "jane.".  We print out
    the other ones we find, which should be the exact four that were present in
    the source. *)
```

Comment:
Maybe we should look for both extensions and attributes?

Reply:
Yes, I misunderstood the point of the test :) It's updated now.